governance/program/src/instruction.rs

```diff
@@ -0,0 +1,264 @@
+    RemoveSignatory,
+
+    /// [Requires Signatory token]
+    /// Adds a Transaction to the Proposal Max of 5 of any Transaction type. More than 5 will throw error.
+    /// Creates a PDA using your authority to be used to later execute the instruction.
+    /// This transaction needs to contain authority to execute the program.
+    ///
+    ///   0. `[writable]` Uninitialized Proposal Transaction account
+    ///   1. `[writable]` Proposal state account
+    ///   2. `[writable]` Signatory account
+    ///   3. `[writable]` Signatory Validation account
+    ///   4. `[]` Proposal account.
+    ///   5. `[]` Governance account
+    ///   6. `[]` Transfer authority
+    ///   7. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   8. `[]` Governance program account
+    ///   9. `[]` Token program account
+    AddCustomSingleSignerTransaction {
+        /// Slot during which this will run
+        delay_slots: u64,
+        /// Instruction
+        instruction: [u8; MAX_PROPOSAL_INSTRUCTION_DATA_LENGTH],
+        /// Position in transaction array
+        position: u8,
+        /// Point in instruction array where 0 padding begins - inclusive, index should be where actual instruction ends, not where 0s begin
+        instruction_end_index: u16,
+    },
+
+    /// [Requires Signatory token]
+    /// Remove Transaction from the Proposal
+    ///
+    ///   0. `[writable]` Proposal State account
+    ///   1. `[writable]` Proposal Transaction account
+    ///   2. `[writable]` Signatory account
+    ///   3. `[writable]` Signatory Validation account
+    ///   5. `[]` Proposal account
+    ///   6. `[]` Transfer Authority
+    ///   7. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   9. `[]` Token program account
+    RemoveTransaction,
+
+    /// [Requires Signatory token]
+    /// Update Transaction slot in the Proposal. Useful during reset periods
+    ///
+    ///   1. `[writable]` Proposal Transaction account
+    ///   2. `[writable]` Signatory account
+    ///   3. `[writable]` Signatory Validation account
+    ///   4. `[]` Proposal State account.
+    ///   5. `[]` Proposal account.
+    ///   6. `[]` Transfer authority.
+    ///   7. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   8. `[]` Token program account
+    UpdateTransactionDelaySlots {
+        /// On what slot this transaction slot will now run
+        delay_slots: u64,
+    },
+
+    /// [Requires Admin token]
+    /// Delete Proposal entirely.
+    ///
+    ///   0. `[writable]` Proposal state account pub key.
+    ///   1. `[writable]` Admin account
+    ///   2. `[writable]` Admin Validation account.
+    ///   3. `[]` Proposal account pub key.
+    ///   4. `[]` Transfer authority.
+    ///   5. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   6. `[]` Token program account.
+    DeleteProposal,
```

Comment:
I removed the Deleted state for now and added Cancelled instead. We might want to keep all the proposals for historical purposes but we also have to keep Proposals in case voters still have source tokens deposited in them.